src/Content/Widget/TrendingTags.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Friendica\Content\Widget;
+
+use Friendica\Core\Cache;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Database\DBA;
+use Friendica\Model\Term;
+
+class TrendingTags
+{
+	/**
+	 * @param string $content 'global' (all posts) or 'local' (this node's posts only)
+	 * @param int    $period  Period in hours to consider posts
+	 * @return string
+	 * @throws \Exception
+	 */
+	public static function getHTML($content = 'global', int $period = 24)
+	{
+		if ($content == 'local') {
+			$tags = self::getLocalTrendingTags($period);
+		} else {
+			$tags = self::getGlobalTrendingTags($period);
+		}
+
+		$tpl = Renderer::getMarkupTemplate('widget/trending_tags.tpl');
+		$o = Renderer::replaceMacros($tpl, [
+			'$title' => L10n::tt('Trending Tags (last %d hour)', 'Trending Tags (last %d hours)', $period),
+			'$more' => L10n::t('More Trending Tags'),
+			'$tags' => $tags,
+		]);
+
+		return $o;
+	}
+
+	/**
+	 * Returns a list of the most frequent global tags over the given period
+	 *
+	 * @param int $period Period in hours to consider posts
+	 * @return array
+	 * @throws \Exception
+	 */
+	private static function getGlobalTrendingTags(int $period)
+	{
+		$tags = Cache::get('global_trending_tags');
+
+		if (!$tags) {
+			$tagsStmt = DBA::p("SELECT t.`term`, COUNT(*) AS `score`
+FROM `term` t
+ JOIN `item` i ON i.`id` = t.`oid` AND i.`uid` = t.`uid`
+ JOIN `thread` ON `thread`.`iid` = i.`id`
+WHERE `thread`.`visible`
+  AND NOT `thread`.`deleted`
+  AND NOT `thread`.`moderated`
+  AND NOT `thread`.`private`
+  AND t.`uid` = 0
+  AND t.`otype` = ?
+  AND t.`type` = ?
+  AND t.`term` != ''
+  AND i.`received` > DATE_SUB(NOW(), INTERVAL ? HOUR)
+GROUP BY `term`
+ORDER BY `score` DESC
+LIMIT 20", Term::OBJECT_TYPE_POST, Term::HASHTAG, $period);
+
+			if (DBA::isResult($tags)) {
+				$tags = DBA::toArray($tagsStmt);
+				Cache::set('global_trending_tags', $tags, Cache::HOUR);
+			}
+		}
+
+		return $tags ?: [];
+	}
+
+	/**
+	 * Returns a list of the most frequent local tags over the given period
+	 *
+	 * @param int $period Period in hours to consider posts
+	 * @return array
+	 * @throws \Exception
+	 */
+	private static function getLocalTrendingTags(int $period)
```

Comment:
This seems for me like a `Friendica\Model\Tags` class method. I don't think we should mix it with Content methods.

Reply:
I would like to have ```Model``` classes only as an interface to the identically named tables.

Means: If it is a ```Model``` thing, then it belongs to the ```Term``` class.

Reply:
That's fine for me too :-)

As a rule of thumb for me, I say `DBA` activities are generally more bound to a `Model` class and `Renderer` activities are generally more bound to a `Content` or `Module` class.